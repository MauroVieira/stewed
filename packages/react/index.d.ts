/**
 * Required to avoid errors when importing these types of files
 */

declare module "*.svg?url" {
  const content: string;
  export default content;
}

declare module "*.svg" {
  const content: React.FunctionComponent<React.SVGAttributes<SVGElement>>;
  export default content;
}

declare module "*.md" {
  const content: string;
  export default content;
}

declare module "*.scss" {
  const classes: { [key: string]: string };
  export default classes;
}
