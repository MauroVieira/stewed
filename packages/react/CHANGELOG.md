# @stewed/react

## 0.1.0

### Minor Changes

- a302a43: Update all packages

### Patch Changes

- Updated dependencies [a302a43]
  - @stewed/utilities@0.2.0
  - @stewed/hooks@0.1.0
  - @stewed/tokens@0.2.0
