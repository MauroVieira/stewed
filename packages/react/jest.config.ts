import { jestConfig } from "@harmonix-hub/testing";

module.exports = {
  ...jestConfig(),
  testEnvironment: "jsdom",
  moduleNameMapper: {
    "\\.(css|less|scss|sss|styl)$": "<rootDir>/node_modules/jest-css-modules",
  }
};
