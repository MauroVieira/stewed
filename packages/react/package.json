{
  "name": "@stewed/react",
  "version": "0.0.0",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "sideEffects": false,
  "license": "MIT",
  "files": [
    "dist/**"
  ],
  "scripts": {
    "dev": "npx vite",
    "build": "npx vite build",
    "lint": "TIMING=1 eslint \"src/**/*.ts*\"",
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist"
  },
  "devDependencies": {
    "@stewed/tsconfig": "workspace:*",
    "@stewed/utils": "workspace:*",
    "@stewed/lint": "workspace:*",
    "@types/react": "^18.0.9",
    "@types/react-dom": "^18.0.4",
    "eslint": "^8.15.0",
    "react": "^18.1.0",
    "tsup": "^5.10.1",
    "typescript": "^4.5.3",
    "sass": "^1.58.0",
    "@vitejs/plugin-react": "^1.3.2",
    "vite": "^2.9.15"
  },
  "publishConfig": {
    "access": "public"
  }
}
