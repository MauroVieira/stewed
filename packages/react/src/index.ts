export * from "./accordion";
export * from "./alert";
export * from "./aspect-ratio";
export * from "./avatar";
export * from "./badge";
export * from "./box";
export * from "./button";
export * from "./card";
export * from "./carousel";
export * from "./checkbox";
export * from "./divider";
export * from "./list-box";
export * from "./progress";
export * from "./radio";
export * from "./tabs";
export * from "./tag";
export * from "./text-area";
export * from "./text-field";
export * from "./theme";
export * from "./theme";
export * from "./toggle";
export * from "./typography";

// Utilities
export * from "./types";
