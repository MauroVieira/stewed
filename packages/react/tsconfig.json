{
  "extends": "@harmonix-hub/tsconfig/react.json",
  "compilerOptions": {
    "outDir": "./dist",
    "types": ["vitest/globals"]
  },
  "include": ["custom.d.ts", "index.ts", "./src/**/*", "./src/index.ts"],
  "exclude": ["node_modules", "dist", "src/**/stories", "src/**/__tests__"]
}
