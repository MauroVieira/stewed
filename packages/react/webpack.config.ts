import path from "path";
// Types
import type { Configuration } from "webpack";

const webpackConfig: Configuration = {
  mode: "production",
  entry: "./src/index.ts",
  output: {
    filename: "index.js",
    library: {
      type: "module",
    },
    path: path.resolve(__dirname, "dist"),
  },
  experiments: {
    outputModule: true,
  },
  optimization: {
    minimize: true,
  },
  module: {
    rules: [
      {
        test: /\.s[ac]ss$/i,
        use: [
          "style-loader",
          {
            loader: "css-loader",
            options: {
              modules: {
                auto: true,
                localIdentName: "[hash:base64]",
              },
              sourceMap: true,
            },
          },
          "sass-loader",
        ],
      },
      {
        test: /\.tsx?$/i,
        exclude: /node_modules/,
        use: {
          loader: "ts-loader",
          options: {
            compilerOptions: {
              sourceMap: true,
            },
          },
        },
      },
    ],
  },
  resolve: {
    extensions: [".tsx", ".ts", ".js"],
  }
};

export default webpackConfig;
