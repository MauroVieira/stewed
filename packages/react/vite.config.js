import { defineConfig } from "vite";
// Node
import { resolve } from "path";
// Plugins
import dts from "vite-plugin-dts";
import peerDepsExternal from "rollup-plugin-peer-deps-external";
import react from "@vitejs/plugin-react";
import { libInjectCss } from "vite-plugin-lib-inject-css";

export default defineConfig({
  plugins: [
    react(),
    libInjectCss(),
    dts({
      copyDtsFiles: true
    }),
  ],
  build: {
    copyPublicDir: false,
    lib: {
      entry: resolve(__dirname, "src/index.ts"),
      formats: ["es"],
    },
    rollupOptions: {
      plugins: [peerDepsExternal()],
      output: {
        assetFileNames: "assets/[name][extname]",
        entryFileNames: "[name].js",
      },
    },
  },
});
