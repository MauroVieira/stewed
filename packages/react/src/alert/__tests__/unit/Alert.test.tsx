import React from "react";
// UI Components
import { Alert, type AlertProps } from "../../index";
// Utilities
import { render } from "@testing-library/react";

describe("Alert", () => {
  describe("Snapshots", () => {
    it("should renders default component", () => {
      const { container } = render(<Alert>Alert</Alert>);

      expect(container).toMatchSnapshot();
    });

    it("should renders title", () => {
      const { container } = render(<Alert title="Title">Alert</Alert>);

      expect(container).toMatchSnapshot();
    });

    it("should render additional classes", () => {
      const { container } = render(<Alert className="other-class">Alert</Alert>);

      expect(container).toMatchSnapshot();
    });

    it.each<AlertProps["shadow"]>(["none", "sm", "md", "lg", "xl", "2xl", "3xl"])(
      "should apply '%s' shadow classes",
      (shadow) => {
        const { container } = render(<Alert shadow={shadow}>Alert</Alert>);

        expect(container).toMatchSnapshot();
      },
    );

    it.each<AlertProps["skin"]>(["primary", "secondary", "neutral", "critical", "success"])(
      "should apply '%s' skin classes",
      (skin) => {
        const { container } = render(<Alert skin={skin}>Alert</Alert>);

        expect(container).toMatchSnapshot();
      },
    );

    it("should render left slot content", () => {
      const { container } = render(<Alert leftSlot="Left slot">Alert</Alert>);

      expect(container).toMatchSnapshot();
    });

    it("should render right slot content", () => {
      const { container } = render(<Alert rightSlot="Right slot">Alert</Alert>);

      expect(container).toMatchSnapshot();
    });

    it("should applies correct size classes", () => {
      const { container } = render(<Alert size="md">Alert</Alert>);

      expect(container).toMatchSnapshot();
    });
  });
});
