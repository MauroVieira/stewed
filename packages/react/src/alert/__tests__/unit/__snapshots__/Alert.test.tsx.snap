// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert > Snapshots > should applies correct size classes 1`] = `
<div>
  <div
    class="_alert_102996 _alert--primary_102996 _alert--md_102996 _alert--shadow-none_102996"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should apply '2xl' shadow classes 1`] = `
<div>
  <div
    class="_alert_102996 _alert--primary_102996 _alert--md_102996 _alert--shadow-2xl_102996"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should apply '3xl' shadow classes 1`] = `
<div>
  <div
    class="_alert_102996 _alert--primary_102996 _alert--md_102996 _alert--shadow-3xl_102996"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should apply 'critical' skin classes 1`] = `
<div>
  <div
    class="_alert_102996 _alert--critical_102996 _alert--md_102996 _alert--shadow-none_102996"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should apply 'lg' shadow classes 1`] = `
<div>
  <div
    class="_alert_102996 _alert--primary_102996 _alert--md_102996 _alert--shadow-lg_102996"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should apply 'md' shadow classes 1`] = `
<div>
  <div
    class="_alert_102996 _alert--primary_102996 _alert--md_102996 _alert--shadow-md_102996"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should apply 'neutral' skin classes 1`] = `
<div>
  <div
    class="_alert_102996 _alert--neutral_102996 _alert--md_102996 _alert--shadow-none_102996"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should apply 'none' shadow classes 1`] = `
<div>
  <div
    class="_alert_102996 _alert--primary_102996 _alert--md_102996 _alert--shadow-none_102996"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should apply 'primary' skin classes 1`] = `
<div>
  <div
    class="_alert_102996 _alert--primary_102996 _alert--md_102996 _alert--shadow-none_102996"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should apply 'secondary' skin classes 1`] = `
<div>
  <div
    class="_alert_102996 _alert--secondary_102996 _alert--md_102996 _alert--shadow-none_102996"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should apply 'sm' shadow classes 1`] = `
<div>
  <div
    class="_alert_102996 _alert--primary_102996 _alert--md_102996 _alert--shadow-sm_102996"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should apply 'success' skin classes 1`] = `
<div>
  <div
    class="_alert_102996 _alert--success_102996 _alert--md_102996 _alert--shadow-none_102996"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should apply 'xl' shadow classes 1`] = `
<div>
  <div
    class="_alert_102996 _alert--primary_102996 _alert--md_102996 _alert--shadow-xl_102996"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should render additional classes 1`] = `
<div>
  <div
    class="_alert_102996 _alert--primary_102996 _alert--md_102996 _alert--shadow-none_102996 other-class"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should render left slot content 1`] = `
<div>
  <div
    class="_alert_102996 _alert--primary_102996 _alert--md_102996 _alert--shadow-none_102996"
    role="alert"
  >
    <div
      class="_alert__left_102996"
    >
      Left slot
    </div>
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should render right slot content 1`] = `
<div>
  <div
    class="_alert_102996 _alert--primary_102996 _alert--md_102996 _alert--shadow-none_102996"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
    <div
      class="_alert__right_102996"
    >
      Right slot
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should renders default component 1`] = `
<div>
  <div
    class="_alert_102996 _alert--primary_102996 _alert--md_102996 _alert--shadow-none_102996"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert > Snapshots > should renders title 1`] = `
<div>
  <div
    class="_alert_102996 _alert--primary_102996 _alert--md_102996 _alert--shadow-none_102996"
    role="alert"
  >
    <div
      class="_alert__wrapper_102996"
    >
      <div
        class="_alert__title_102996"
      >
        Title
      </div>
      <div
        class="_alert__body_102996"
      >
        Alert
      </div>
    </div>
  </div>
</div>
`;
